controllers/front/callbackok.php

```diff
@@ -50,13 +48,13 @@ public function postProcess()
             }
 
             // Handle success
-            if ($response->wasSuccessful()) {
+            if ($response) {
                 $orderStatus = (int) Configuration::get('PS_OS_PAYMENT');
-                $paymentType = $response->getPrimaryPayment()->getAuthType();
-                $amountPaid = $response->getPrimaryPayment()->getCapturedAmount();
-                $captureStatus = Tools::getValue('require_capture');
-                $currencyPaid = Currency::getIdByIsoCode($response->getPrimaryPayment()->getCurrency());
-                $transactionID = Tools::getValue('transaction_id');
+                $paymentType = $response->type;
+                $amountPaid = $response->Transactions[0]->CapturedAmount;
```

Comment:
We should probably check if there are any transactions before accessing the array.